python-backend/tests/permit/resources/test_permit_resource.py

```diff
@@ -1,29 +1,143 @@
+        mine_2_permit_guid=MINE_2_PERMIT_GUID,
+        bad_guid=NON_EXISTENT_GUID,
+        in_use_permit_no=test_permit_no)
+
+    db.session.query(PermitAmendment).delete()
+    db.session.commit()
+    db.session.query(Permit).delete()
+    db.session.commit()
+    db.session.delete(mine_1)
+    db.session.delete(mine_2)
+    db.session.commit()
 
 
 # GET
-def test_get_permit_not_found(test_client, auth_headers):
-    get_resp = test_client.get('/permits/' + TEST_MINE_GUID, headers=auth_headers['full_auth_header'])
+def test_get_permit_not_found(test_client, setup_info, auth_headers):
+    get_resp = test_client.get(
+        '/permits/' + setup_info.get('bad_guid'), headers=auth_headers['full_auth_header'])
     get_data = json.loads(get_resp.data.decode())
-    assert get_data == {
-        'error': {
-            'status': 404,
-            'message': 'Permit not found'
-        }
-    }
+    assert get_data == {'error': {'status': 404, 'message': 'Permit not found'}}
     assert get_resp.status_code == 404
 
 
-def test_get_permit(test_client, auth_headers):
-    get_resp = test_client.get('/permits/' + TEST_PERMIT_GUID_1, headers=auth_headers['full_auth_header'])
+def test_get_permit(test_client, setup_info, auth_headers):
+    get_resp = test_client.get(
+        '/permits/' + setup_info.get('mine_2_permit_guid'),
+        headers=auth_headers['full_auth_header'])
     get_data = json.loads(get_resp.data.decode())
-    assert get_data['permit_guid'] == TEST_PERMIT_GUID_1
+    assert get_data['permit_guid'] == setup_info.get('mine_2_permit_guid')
     assert get_resp.status_code == 200
 
 
-def test_get_mine_by_mine_guid(test_client, auth_headers):
-    get_resp = test_client.get('/mines/' + TEST_MINE_GUID, headers=auth_headers['full_auth_header'])
-    get_data = json.loads(get_resp.data.decode())
-    assert TEST_PERMIT_GUID_1 in list(map(lambda x: x['permit_guid'], get_data['mine_permit']))
-    assert get_resp.status_code == 200
+def test_get_with_permit_no(test_client, setup_info, auth_headers):
+    get_resp = test_client.get(
+        '/permits?permit-no=' + setup_info.get('in_use_permit_no'),
+        headers=auth_headers['full_auth_header'])
+    assert get_resp.status_code == 400
+
+
+#Create
+def test_post_permit(test_client, setup_info, auth_headers):
+    PERMIT_NO = 'mx-test-999'
+    data = {
+        'mine_guid': setup_info.get('mine_1_guid'),
+        'permit_no': PERMIT_NO,
+        'permit_status_code': 'O',
+        'received_date': '1999-12-12',
+        'issue_date': '1999-12-21',
+        'authorization_end_date': '2012-12-02'
+    }
+    post_resp = test_client.post('/permits', headers=auth_headers['full_auth_header'], data=data)
+    post_data = json.loads(post_resp.data.decode())
+
+    assert post_resp.status_code == 200
+    assert post_data.get('mine_guid') == setup_info.get('mine_1_guid')
+    assert post_data.get('permit_no') == PERMIT_NO
+    assert post_data.get('amendments')
```

Comment:
None and empty list are both falsy, what does this assert catch?

Reply:
I will remove the additional check. I just wanted to assert that there was only one amendment.

Reply:
I liked the explicit len(post_data.get('amendments')) > 0 (or == 1). thought the generic falsy check should go.

Reply:
K, will revert back.